src/distPlus/distribution/XYShape.re

```diff
@@ -227,15 +283,16 @@ module PointwiseCombination = {
           outX.push(x);
           outY.push(fn(ya, yb));
         }
+
         return {xs: outX, ys: outY};
       }
     |}];
 
-  let combine =
+  let combineEvenXs =
```

Comment:
maybe combineXsEvenlyDivided or similar?
I originally thought you meant that only even samples/points were used.